ngraph/core/src/op/roi_pooling.cpp

```diff
@@ -36,32 +36,81 @@ op::ROIPooling::ROIPooling(const Output<Node>& input,
 
 void op::ROIPooling::validate_and_infer_types()
 {
-    auto input_et = get_input_element_type(0);
+    auto feat_maps_et = get_input_element_type(0);
+    auto coords_et = get_input_element_type(1);
+    NODE_VALIDATION_CHECK(
+        this,
+        feat_maps_et.is_real() && coords_et.is_real(),
+        "The data type for input and ROIs is expected to be a floating point type. Got: ",
+        feat_maps_et,
+        " and: ",
+        coords_et);
+
+    NODE_VALIDATION_CHECK(
+        this,
+        feat_maps_et == coords_et,
+        "Type of feature maps (inputs) and rois is expected to be the same. Got: ",
+        feat_maps_et,
+        " and: ",
+        coords_et);
+
+    NODE_VALIDATION_CHECK(
+        this,
+        m_spatial_scale > 0,
+        "The spatial scale attribute should be a positive floating point number. Got: ",
+        m_spatial_scale);
+
+    NODE_VALIDATION_CHECK(
+        this,
+        m_method == "max" || m_method == "bilinear",
+        "Pooling method attribute should be either \'max\' or \'bilinear\'. Got: ",
+        m_method);
+
     if (get_input_partial_shape(0).is_static() && get_input_partial_shape(1).is_static())
```

Comment:
This check is too rough. We can calculate some of the output dimensions if `get_input_partial_shape(0)` or `get_input_partial_shape(1)` has **some** dynamic dimensions. Please, make the shape inference function more specific and try to calculate as much output dimensions as possible. And cover these changes with several shape inference unit tests.

Reply:
Done